artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireMessageConverter.java

```diff
@@ -590,9 +591,11 @@ private static ActiveMQMessage toAMQMessage(MessageReference reference,
       }
       amqMsg.setCommandId(commandId);
 
-      final SimpleString corrId = getObjectProperty(coreMessage, SimpleString.class, OpenWireConstants.JMS_CORRELATION_ID_PROPERTY);
-      if (corrId != null) {
-         amqMsg.setCorrelationId(corrId.toString());
+      final Object correlationID = coreMessage.getCorrelationID();
+      if (correlationID instanceof String || correlationID instanceof SimpleString) {
+         amqMsg.setCorrelationId(correlationID.toString());
+      } else if (correlationID instanceof byte[]) {
+         amqMsg.setCorrelationId(new String((byte[])correlationID, StandardCharsets.UTF_8));
```

Comment:
I thought about this previously, and I tried to come up with a test to break it. However, I couldn't come up with "not actually UTF-8 decodable content." Do you have an example?

Reply:
You could send bytes for a normal two byte encoding where the second byte violates the prefix value of the encoding rules.  The UTF8 spec uses upper bits of the bytes to stretch out the characters encodings across subsequent bytes, looking into the proton-j read UTF8 reading code in ReadableBuffer would likely shed a bit of light on how to corrupt a trailing encoded byte.

Reply:
After doing some more testing directly with `new String(byte[])` I see now why I wasn't able to generate a failure. This is what the JavaDoc for that method states (in part):

> This method always replaces malformed-input and unmappable-character sequences with this charset's default replacement string. 

So this won't blow up and tank the message conversion even with "not actually UTF-8 decodable content."

Reply:
Yeah I forgot about that nugget; so instead of blowing up or sending nothing it will just send a corrupted representation of the original bytes. Arguably worse? :)

Per the next bit of that javadoc, you'd need to actually decode the bytes as UTF-8 rather than just creating a String instance with them, so that the decoder would instead blow up for garbage rather than substitute. Tim's idea for malformed bytes is simple, but as per your link the simplest is actually just a single byte that isnt an ASCII char or the multi-byte prefix.

Reply:
It will now return `null` for correlation ID when the bytes are invalid UTF-8 and log a `WARN`.